File: Q1.py
número = int(input("Digite um número maior do que 0: "))
if número > 0:
    ordem = int(input("Digite 1 para ordem crescente ou 2 para ordem decrescente: "))
    if ordem == 1:
        número -= 1
        print("A ordem crescente do zero até este número é ", end = '')
        for listagem in range(0, número):
            print(f"{listagem}, ", end = '')
        print(f"{número} e {número + 1}.")
    elif ordem == 2:
        print("A ordem decrescente deste número até zero é ", end = '')
        for listagem in range(0, número - 1):
            listagem = número - listagem
            print(f"{listagem}, ", end = '')
        print(f"1 e 0.")
    else:
        print("Você deve digitar um dos valores pedidos.")
else:
    print("Você deve digitar um número maior do que 0.")

File: Q1.cpp
#include <iostream>
#include <stdio.h>

using namespace std;

int main()
{
    int numero, ordem;
    cout << "Digite um número maior do que 0: ";
    cin >> numero;
    if (numero > 0)
    {
        cout << "Digite 1 para ordem crescente ou 2 para ordem decrescente: ";
        cin >> ordem;
        if (ordem == 1)
        {
            numero --;
            cout << "A ordem crescente do zero até este número é ";
            for (int listagem = 0; listagem < numero; listagem ++)
            {
                printf("%d, ", listagem);
            }
            printf("%d e %d.\n", numero, numero + 1);
        }
        else if (ordem == 2)
        {
            cout << "A ordem decrescente deste número até zero é ";
            for (int listagem = numero; listagem > 1; listagem --)
            {
                printf("%d, ", listagem);
            }
            cout << "1 e 0." << endl;
        }
        else
        {
            cout << "Você deve digitar um dos valores pedidos." << endl;
        }
    }
    else
    {
        cout << "Você deve digitar um número maior do que 0." << endl;
    }
    return 0;
}